include/experimental/__p0009_bits/mdspan.hpp

```diff
@@ -111,40 +115,78 @@ class mdspan
   //--------------------------------------------------------------------------------
   // [mdspan.basic.cons], mdspan constructors, assignment, and destructor
 
-  MDSPAN_INLINE_FUNCTION_DEFAULTED constexpr mdspan() = default;
+#if !MDSPAN_HAS_CXX_20
+  MDSPAN_FUNCTION_REQUIRES(
+    (MDSPAN_INLINE_FUNCTION constexpr),
+    mdspan, , ,
+    /* requires */ (
+       (rank_dynamic() > 0) &&
+       _MDSPAN_TRAIT(is_default_constructible, pointer) &&
+       _MDSPAN_TRAIT(is_default_constructible, mapping_type) &&
+       _MDSPAN_TRAIT(is_default_constructible, accessor_type)
+     )
+   )
```

Comment:
Does `= default` not work here?  I was wondering why you needed `MDSPAN_HAS_CXX_20`.

Reply:
How do we attach sfinae requirements to implement the constraints on a non-tempalted default constructor without C++20 requires?

Reply:
Ah, yes, that sounds hard.  It's probably best not to worry about it in the back-port.  Otherwise you would have to specialize the whole class just for the default constructor.